lib/stat_tracker.rb

```diff
@@ -26,72 +22,43 @@ def initialize(games_data, teams_data, game_teams_data)
   end
 
   def highest_total_score
-    @games_data.max_by { |game| game.total_score}.total_score
+    @games_data.max_by(&:total_score).total_score
   end
 
   def lowest_total_score
-    @games_data.min_by { |game| game.total_score}.total_score
+    @games_data.min_by(&:total_score).total_score
   end
 
   def biggest_blowout
-    @games_data.max_by { |game| game.margin_of_victory }.margin_of_victory
+    @games_data.max_by(&:margin_of_victory).margin_of_victory
   end
 
   def percentage_home_wins
-    # Push home wins definition to game data class method & call here?
-    home_wins = @games_data.select{ |game| game.home_goals > game.away_goals}.length
-    average(home_wins, @games_data.length).round(2)
+    average(Game.home_wins, @games_data.length).round(2)
   end
 
   def percentage_visitor_wins
-    # Push visitor wins definition to game data class method & call here?
-    visitor_wins =  @games_data.select{ |game| game.away_goals > game.home_goals}.length
-    average(visitor_wins, @games_data.length).round(2)
+    average(Game.visitor_wins, @games_data.length).round(2)
   end
 
   def percentage_ties
-    # Push ties definition to game data class method & call here?
-    ties = @games_data.select{ |game| game.margin_of_victory == 0}.length
-    average(ties, @games_data.length).round(2)
+    average(Game.ties, @games_data.length).round(2)
   end
 
   def count_of_games_by_season
-    # Push all this logic to class method in Game & call that here?
-    @games_data.reduce({}) do |games_count, game|
-      if games_count[game.season.to_s].nil?
-        games_count[game.season.to_s] = 1
-      else
-        games_count[game.season.to_s] += 1
-      end
-      games_count
-    end
+    Game.count_of_games_by_season
```

Comment:
The logic for this method now lives as a class method on the Games class. I refactored there as well, reducing the lines of code to 2.